File: Packages/NotebarCore/Sources/NotebarCore/Models/Theme.swift
import Foundation

/// The user's appearance preference (spec §6.5): a three-way choice between
/// following macOS's own appearance and pinning to one of its two modes.
///
/// The `NSAppearance` mapping deliberately does not live here — this module
/// never imports AppKit (spec §3, rule 1) — see the app target's
/// `Theme.nsAppearance` for that half.
public enum Theme: String, CaseIterable, Equatable, Sendable {
    /// Tracks macOS's own appearance live, including its automatic day/night
    /// schedule. The default: an overlay that floats above other apps should
    /// match them.
    case system
    case light
    case dark

    /// What `AppStateRepository.theme()` falls back to when nothing has been
    /// saved yet, or the saved value is no longer recognised.
    public static let `default`: Theme = .system
}

File: windows/Notebar.Core/Models/Theme.cs
namespace Notebar.Core.Models;

/// <summary>The user's appearance preference: a three-way choice between
/// following Windows' own appearance and pinning to one of its two modes.</summary>
public enum Theme { System, Light, Dark }

public static class ThemeExtensions
{
    /// <summary>The default: an overlay that floats above other apps should
    /// match them.</summary>
    public static Theme Default => Theme.System;

    /// <summary>Falls back to <see cref="Default"/> both when nothing has been
    /// saved and when the saved value is no longer recognised — someone
    /// hand-editing the database must get the default appearance back, not a
    /// crash.</summary>
    public static Theme Parse(string? raw) => raw switch
    {
        "light" => Theme.Light,
        "dark" => Theme.Dark,
        "system" => Theme.System,
        _ => Default,
    };

    public static string ToStorageString(this Theme theme) => theme switch
    {
        Theme.Light => "light",
        Theme.Dark => "dark",
        _ => "system",
    };
}
